privval/file.go

```diff
@@ -184,8 +183,12 @@ func NewFilePV(privKey crypto.PrivKey, keyFilePath, stateFilePath string) *FileP
 
 // GenFilePV generates a new validator with randomly generated private key
 // and sets the filePaths, but does not call Save().
-func GenFilePV(keyFilePath, stateFilePath string) *FilePV {
-	return NewFilePV(ed25519.GenPrivKey(), keyFilePath, stateFilePath)
+func GenFilePV(keyFilePath, stateFilePath string, keyGen func() (crypto.PrivKey, error)) *FilePV {
```

Comment:
I think we could create a new method, receiving this third parameter. Then we maintain this method as it is, but calling the new method with the default implementation, i.e., ed255519.

This in particular allows us to not change the `cmd/priv_val_server` at all.